sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/OutputBinding/EventGridAsyncCollector.cs

```diff
@@ -51,128 +51,83 @@ public async Task FlushAsync(CancellationToken cancellationToken = default(Cance
             {
                 // determine the schema by inspecting the first event (a topic can only support a single schema)
                 var firstEvent = events.First();
-                if (firstEvent is string str)
+                switch (firstEvent)
                 {
-                    bool isEventGridEvent = false;
-                    try
-                    {
-                        var ev = EventGridEvent.Parse(new BinaryData(str));
-                        isEventGridEvent = true;
-                    }
-                    catch (ArgumentException)
-                    {
-                    }
-
-                    if (isEventGridEvent)
+                    case string:
```

Comment:
```suggestion
                    case string s:
```

Reply:
to avoid the extra cast

Reply:
You can also pull await SendAsync out of the switch

```
BinaryData bd = evt switch {
...
};
```

Reply:
Actually, that is just the first event, so we still need to do the cast for the other events.